### ISSUE TITLE
lwt backend without a dep on `tls`?

### ISSUE THREAD

**Issue by c-cube** [mirage/capnp-rpc#187]
tls pulls a lot of dependencies, it'd be very convenient to just use normal lwt networking functions.

**Comment by avsm** [comment]
To be clear: you're looking for a cleartext-on-the-wire Lwt version?

This is possible, but a lot of engineering just for the case of dropping some dependencies in return for possibly insecure deployments.  Could you elaborate a little more about your usecase?

**Comment by c-cube** [comment]
well, I was exploring options for IPC on a single machine, but capnp seems way overkill for that.

**Comment by talex5** [comment]
Yes, it's probably overkill if you can't afford the cost of a TLS library.

For communication within a machine, you might prefer [ocaml-wayland](https://github.com/talex5/ocaml-wayland) (which is typically used for graphics stuff, but doesn't have to be).
